### PR TITLE
Minor UI changes are remaining

### PR DESCRIPTION
**Pull request by DanishOnGit**

### CODE REVIEW

File: src/Components/ProductCard.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          <span className="line-through small">Rs{product.price} </span>
+          <span className="discount">{product.offer}% OFF</span>
+        </p>
+        <button
+          disabled={!product.inStock || isDisabled}
+          onClick={() => addToCartHandler(cartItems, dispatch, product,showToast,setIsDisabled,isRendered)}
+          className={
+            product.inStock
+              ? "btn btn-primary add-to-cart-btn"
+              : "btn btn-primary add-to-cart-btn disabled"
```
Comment (ritikarp) [inline src/Components/ProductCard.js:74]: The style for out of stock product card is not visible or clear at the first glance, here's what you can do:
- remove the box-shadow effect on hover 
- add an overlay text for out of stock
- change the button text, don't show add to cart
- set cursor to not-allowed

File: src/Components/Wishlist.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                <div className="product-details-wrapper">
+                  <p className="strong">{item.brand}</p>
+                  <p className="offer-wrapper">
+                    <span className="strong">Rs{getFinalPrice(item.price,item.offer)}</span>
+                    <span className="line-through">Rs. {item.price} </span>
+                    <span className="discount">{item.offer}% OFF</span>
+                  </p>
+                </div>
+              </div>
+              <button
```
Comment (ritikarp) [inline src/Components/Wishlist.js:67]: A product out of stock, when added to the wishlist, can be added to the cart from here.
Suggestions to correct this,
- change the text to Notify me from Move to cart
- disable the button

File: src/Components/ProductsListingPage.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    return productsList.length;
+  }
+
+  return (
+    <div>
+      <Toast />
+      <h1 className="product-listing-page-header centered">
+        All Products {totalNumberOfProducts(productsList)}
+      </h1>
+      <div className="product-listing-page-wrapper">
```
Comment (ritikarp) [inline src/Components/ProductsListingPage.js:86]: Add some margin for this component.

File: src/Components/Navbar.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  } = useCart();
+
+  function totalCartItems(cartItems) {
+    const array = cartItems.filter((product) => product.existsInCart);
+    return array.length;
+  }
+
+  return (
+    <div>
+      <nav className="nav-wrapper-3">
```
Comment (ritikarp) [inline src/Components/Navbar.js:17]: You can work on your navbar styling as it appears consistently on each page:
- remove the underline from logo and product anchor tags. 
- Replace logo with "Shuttle Shop" to give it a complete look.

**Review by ritikarp — CHANGES_REQUESTED** [review]
Hey Danish @DanishOnGit 
Kudos for completing your e-commerce app. 

I have added some remarks for your code, have a look. I hope the comments will be helpful. Rest LGTM!

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    
+  }, []);
+
+  return (
+    <div className="App">
+      <Navbar searchText={searchText} setSearchText={setSearchText} />
+{/* 
+      {loading && (
+        <Loader type="ThreeDots" color="#fc452e" height={80} width={80} />
+      )} */}
```
Comment (DarkShadow20) [inline src/App.js:69]: If this is not needed you can delete it.

File: src/App.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    dispatch
+  } = useCart();
+
+  useEffect(() => {
+    (async function () {
+      try {
+        setLoading(true);
+        const response = await axios.get("./api/products");
+        dispatch({ type: "GET_PRODUCTSLIST", payload: response.data.products });
+        // setProductsList(response.data.products);
```
Comment (DarkShadow20) [inline src/App.js:32]: You should remove comments from your code if not needed.

File: src/Components/Cart.js [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+  }
+
+  function movetowishlist(item) {
+    removeFromCartHandler(item);
+    const response = checkIfAlreadyPresent(wishlistItems, item.id);
+    if (!response) {
+      wishlistHandler(wishlistItems, dispatch, item);
+    } else if (response && response.existsInWishlist === false) {
+      wishlistHandler(wishlistItems, dispatch, item);
+    }
```
Comment (DarkShadow20) [inline src/Components/Cart.js:36]: Just a good practice that you should use else at end of else if conditions. If wrong correct me,

**Comment by DarkShadow20** [comment]
Danish code is very well written. Suggested some minor code consistency.

File: src/App.js (hunk 6 above)
Comment (DanishOnGit) [reply src/App.js:32]: Yes will do that

File: src/Components/Cart.js (hunk 7 above)
Comment (DanishOnGit) [reply src/Components/Cart.js:36]: No Kunal . We should use an "else" only if we wish to do something if the " if " condn is not satisfied.
Here, we want to call wishlistHandler only if  any one of the two conditions is met. And if nothing is satisfied then we don't  want to do anything, hence I avoided writing "else" .

File: src/App.js (hunk 5 above)
Comment (DanishOnGit) [reply src/App.js:69]: Thanks Kunal !
Yes.. I am planning to exclude the loader now.

File: src/Components/Wishlist.js (hunk 2 above)
Comment (DanishOnGit) [reply src/Components/Wishlist.js:67]: Yes! I will fix this for sure.

File: src/Components/ProductCard.js (hunk 1 above)
Comment (DanishOnGit) [reply src/Components/ProductCard.js:74]: Thanks Ritika !
Yes even I feel the out of stock card is badly done. I will try to add an overlay and remove the hover effect

File: src/Components/Cart.js (hunk 7 above)
Comment (DanishOnGit) [reply src/Components/Cart.js:36]: No Kunal . We should use an "else" only if we wish to do something if the " if " condition is not satisfied.
Here, we want to call wishlistHandler only if any one of the two conditions is met. And if nothing is satisfied then we don't want to do anything, hence I avoided writing "else" .

**Review by gkanishk — COMMENTED** [review]
LGTM!

**Pull request merged**